### PR TITLE
[F2F-666] Emit TxMA Events in IPR

### PR DESCRIPTION
**Pull request by Caitlin-cooling**
### What changed
- Add `sendToTXMA` method to IPRService (copied from F2F)
- Added `TxMAUtils` including types and `buildCoreEventFields` function 

### Evidence

Successful event sent to TxMA queue from `SendEmailProcessor`
Screenshot 2023-06-20 at 1 43 40 pm

### Why did it change

We need to emit events to TxMA to say that we have emailed a user with their result and they have accessed their result

### Issue tracking
- [F2F-666](https://govukverify.atlassian.net/browse/F2F-666)

### PII logging

- [x] Verified that no PII data is being logged

### Other considerations

- We were not able to test the changes to the `SessionProcessor` as we could not trigger it without a frontend. It has been unit tested and the other lambda has been tested so we can be fairly confident that it works here and will test it in dev after merge

[F2F-666]: https://govukverify.atlassian.net/browse/F2F-666?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/services/SendEmailProcessor.ts [hunk 1]
```diff
@@ -1,46 +1,61 @@
...
-    	await this.validationHelper.validateModel(email, this.logger);
+		await this.validationHelper.validateModel(email, this.logger);
+
+		const emailResponse: EmailResponse = await this.govNotifyService.sendEmail(email);
 
-    	const emailResponse: EmailResponse = await this.govNotifyService.sendEmail(email);
-    	this.logger.debug("Response after sending Email message", { emailResponse });
+		await this.iprService.sendToTXMA({
+			event_name: "IPR_RESULT_NOTIFICATION_EMAILED",
+			...buildCoreEventFields({}),
```
Comment (richa-misra) [inline src/services/SendEmailProcessor.ts:54]: could you check with Sanka please if adding email address as per [this RFC](https://github.com/alphagov/digital-identity-architecture/blob/main/rfc/0023-crossteam-audit-and-analysis-eventmessage-payload.md) for IPR_RESULT_NOTIFICATION_EMAILED event would be useful

**Review by richa-misra — DISMISSED** [review]

File: src/services/SendEmailProcessor.ts (hunk 1 above)
Comment (Caitlin-cooling) [reply src/services/SendEmailProcessor.ts:54]: Sure, asked him

Comment (Caitlin-cooling) [reply src/services/SendEmailProcessor.ts]: He has confirmed we want email so I've added that. He has also confirmed that we need user_id so having a discussion offline about how we might get access to that here

Comment (Caitlin-cooling) [reply src/services/SendEmailProcessor.ts]: Agreed that user_id will come later as there is more work required to get it

**Review by richa-misra — APPROVED** [review]

**Pull request merged**